packages/vitest/src/node/core.ts

```diff
@@ -83,12 +87,22 @@ export class Vitest {
 
     this.reporters = await createReporters(resolved.reporters, this.runner)
 
+    if (options.coverage?.enabled && options.coverage?.provider === 'c8')
+      this.coverageReporter = new C8Reporter()
+    else if (options.coverage?.enabled && options.coverage.provider === 'istanbul')
+      this.coverageReporter = new IstanbulReporter()
+    else
+      this.coverageReporter = new NullCoverageProvider()
```

Comment:
Move this to some `getCoverageProvider(options.coverage)` method in `../integrations/coverage`

Off topic: Should this feature be named coverage reporters or coverage providers?

Reply:
I think we should keep naming these as coverage providers - at least internally in the codebase. I'll rename all new coverage reporters to providers.

Coverage reporters are then the ones `istanbul-repoters` (also used by `c8` internally) support, e.g. `lcov`, `clover`.

What Jest has is [`coverageProviders` configuration option](https://jestjs.io/docs/configuration#coverageprovider-string) with values of `babel` and `v8`. They also expose only a few coverage related options, instead of the whole `c8` API. Both coverage providers use the same public API. If we aim to do the same, that is a breaking change and requires some more work.

Reply:
I don't think we need to make c8 and Istanbul options the same. Proper types for options as it is now is enough in my opinion.

Reply:
Changes done in
- https://github.com/vitest-dev/vitest/compare/ebb21c001a5ea389ad610dee931f4ee16c6be422..fea9cce8ec88641a632de524d46a833dff336769
- https://github.com/vitest-dev/vitest/compare/fea9cce8ec88641a632de524d46a833dff336769..fc1375028b0f44ec8ce16b71f2c602c8a5af3375